lib/common/output_text.c

```diff
@@ -73,6 +76,36 @@ text_reset(pcmk__output_fmt_t *out) {
     text_init(out);
 }
 
+static void
+text_register_message(pcmk__output_fmt_t *out, const char *message_id,
+                      pcmk__message_fn_t fn) {
+    text_private_t *priv = out->priv;
+
+    CRM_ASSERT(priv != NULL);
+
+    g_hash_table_replace(priv->messages, strdup(message_id), fn);
+}
+
+static bool
+text_message(pcmk__output_fmt_t *out, const char *message_id, ...) {
+    va_list args;
+    text_private_t *priv = out->priv;
+    pcmk__message_fn_t fn;
+
+    CRM_ASSERT(priv != NULL);
+
+    fn = g_hash_table_lookup(priv->messages, message_id);
+    if (fn == NULL) {
+        return false;
+    }
+
+    va_start(args, message_id);
+    fn(out, args);
+    va_end(args);
+
+    return true;
+}
+
```

Comment:
Unless I'm missing something, I don't think register_message() and message() need to be different per output format. There's no problem with reusing a function for multiple formats (we'll definitely do that with html and cgi), or we can pull these two functions out of the struct.